handler/src/main/java/io/netty/handler/ssl/SslContext.java

```diff
@@ -93,6 +93,8 @@
         }
     }
 
+    final boolean startTls;
```

Comment:
@normanmaurer any feedback on that?
After some thought it wouldn't make sense to do the try/finally in SslContext.newHandler since it is only useful when adding the handler to a pipeline.

What would be best is:

- remove SslContext.newHandler
- In SniHandler.replaceHandler, directly use sslContext.newHandler (not the static one), so no need for visible startTls and no reference to the engine is held in SniHandler.replaceHandler.
- have a way to release the engine even though we don't have a reference to it in SniHandler.replaceHandler: I don't know if calling ReferenceCountUtil.safeRelease(sslHandler) will release the engine too…

what do you think?